src/schema/browser-apis.js

```diff
@@ -0,0 +1,39 @@
+import { DEPRECATED_APIS, TEMPORARY_APIS } from 'const';
+import schemaList from 'schema/firefox-schemas';
+
+const schemaArrayNames = ['functions', 'events'];
+const schemaObjectNames = ['types', 'properties'];
+const schemas = schemaList.reduce((all, current) => ({
+  ...all,
+  [current.id]: current,
+}), {});
+
+export function hasBrowserApi(namespace, property) {
+  const schema = schemas[namespace];
+  // We "have" the API if it's deprecated or temporary so
+  // we don't double warn.
+  if (isDeprecatedApi(namespace, property)
+    || isTemporaryApi(namespace, property)) {
+    return true;
+  }
+  if (!schema) {
+    return false;
+  }
+  return schemaObjectNames.some((schemaProperty) => {
+    return schema[schemaProperty] && property in schema[schemaProperty];
+  }) || schemaArrayNames.some((schemaProperty) => {
```

Comment:
It might be more readable to separate these out into separate blocks that return.